### PR TITLE
Add Python testing infrastructure

### PR DESCRIPTION
**Pull request by jkfran**
## Done

- Have a `tests` folder
- Have at least one test in the `tests` folder, based on `unittest`, to test that the `/` homepage returns a `200` response
- `./run test` run `lint-python` and `test-python`
- `./run lint-python` run `flake8` and `black --check --line-length 79` against all Python code (including tests)
- `./run test-python` run the `tests`
- Set up CircleCI with a job called `test-python`, which run the tests
- Added codecov integration on CircleCI

## QA

- Check out this feature branch
- Run the command `./run test`, it should end correctly.
- Run the command `./run lint-python`, it should end correctly.
- Run the command `./run test-python`, it should end correctly.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -5,48 +5,66 @@ defaults: &defaults
...
+          name: Run tests with coverage
+          command: |
+            coverage run  --source=. -m unittest discover tests
+            bash <(curl -s https://codecov.io/bash) -cF python
 workflows:
   version: 2
   build:
     jobs:
-      - test-site
-      - lint-nginx
```
Comment (nottrobin) [inline .circleci/config.yml]: Did you remove `lint-nginx`? Do you think we don't need it?

**Comment by nottrobin** [comment]
One question ^

File: .circleci/config.yml (hunk 1 above)
Comment (jkfran) [reply .circleci/config.yml]: This project wasn't set up on CircleCI and this was the first time I saw this check. I thought this test was an older one or something. Also is it possible to make changes in the nginx.conf? I thought we were using the deployment configs for this site.

Should I add it back? I was also confused by the command `gixy nginx.conf` Do we have an nginx.conf file in that path?

Comment (nottrobin) [reply .circleci/config.yml]: Ah you're right. We use `nginx.conf`s in our static sites - e.g. https://github.com/canonical-web-and-design/cloud-init.io/blob/master/nginx.conf - and then we use Gixy to test them.

But you're right, this is now a Flask project and doesn't include a, `nginx.conf`. I'll remove that requirement and merge.

**Pull request merged**